### DISCUSSION TITLE
ArrAPI - Kometa

### DISCUSSION THREAD

**Discussion by ezekieldas** [buildarr/buildarr#212] [Ideas]
I just stumbled up this: https://github.com/Kometa-Team/ArrAPI

I'm not yet familiar enough with the buildarr codebase to determine how useful it might be.

**Comment by Callum027** [comment]
Thanks for making me aware of this library.

It's interesting, but as this is a library for interacting with Arr instances on an everyday basis (e.g. adding media to the instance for monitoring/downloading etc), it is different from Buildarr's use case.

With Buildarr, in all plugins I use OpenAPI-generated Python APIs from [DevOpsArr](https://github.com/devopsarr) to interact with the *Arr instances. These libraries smooth out some of the edge cases of interacting with the *Arr APIs (e.g. what to do when a field is non-existent on a retruned JSON response), so they have been extremely helpful and I have helped report issues where I can.

The only plugin that does not use it right now is the Sonarr plugin, as I wrote that before these libraries existed, but [when I add support for Sonarr V4](https://github.com/buildarr/buildarr-sonarr/issues/7) I will officially switch to using them.
